ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/RemoveUnusedNamedPatternFix.fs

```diff
@@ -0,0 +1,75 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open System
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Psi.Util
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.TextControl
+
+type RemoveUnusedNamedPatternFix(warning: UnusedValueWarning) =
+    inherit FSharpQuickFixBase()
+    
+    let fieldAndRecord =
+        match warning.Pat.IgnoreInnerParens().Parent with
+        | :? IFieldPat as fieldPat ->
+            match fieldPat.Parent with
+            | :? IRecordPat as recordPat ->
+                Some(fieldPat, recordPat)
```

Comment:
Please use generated navigation helpers instead of manually traversing the tree: it's harder to make a mistake and it'll make changing the tree structure much easier later, and we often do it for some parts of the tree. `RecordPatNavigator.GetByFieldPattern` is what you'd want to use here.